File: leet-code/938. Range Sum of BST/RangeSumOfBST938.java
package com.example;

public class RangeSumOfBST938 {
    private int rangeSum = 0;

    public int rangeSumBST(TreeNode root, int low, int high) {
        if (root == null) return 0;
        if (low <= root.val && root.val <= high) rangeSum += root.val;
        rangeSumBST(root.left, low, high);
        rangeSumBST(root.right, low, high);
        return rangeSum;
    }
}

File: leet-code/938. Range Sum of BST/RangeSumOfBST938.kt
package com.example

class RangeSumOfBST938 {
    fun rangeSumBST(root: TreeNode?, low: Int, high: Int): Int {
        if (root == null) {
            return 0
        }

        var sum = 0
        sum += rangeSumBST(root.left, low, high)
        sum += rangeSumBST(root.right, low, high)
        if (low <= root.`val` && root.`val` <= high) {
            sum += root.`val`
        }
        return sum
    }
}
